lib/haml_lint/linter/repeated_id.rb

```diff
@@ -0,0 +1,29 @@
+module HamlLint
+  # Detects repeated instances of an element ID in a file
+  class Linter::RepeatedId < Linter
+    include LinterRegistry
+
+    MESSAGE_FORMAT = %{Do not repeat id "#%s" on the page}.freeze
+
+    def visit_tag(node)
+      id = node.tag_id
+      return unless id && !id.empty?
+
+      nodes = (id_map[id] << node)
+      case nodes.size
+      when 2 then nodes.each { |logged_node| add_lint(logged_node, id) }
+      when (3..Float::INFINITY) then add_lint(node, id)
```

Comment:
It took a while for me to make sense of this, but I think I got it now. When the second id is found, add lints to all (two) lines. When the next one is found, just add a lint to that line.